### PR TITLE
feat: receive accepted wholesale services event from wholesale

### PR DESCRIPTION
**Pull request by MadsDue**

### CODE REVIEW

File: source/Process.Application/Transactions/WholesaleServices/WholesaleServicesRequestAcceptedMapper.cs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        return (missing: quantityQualities.Contains(QuantityQuality.Missing),
+                estimated: quantityQualities.Contains(QuantityQuality.Estimated),
+                measured: quantityQualities.Contains(QuantityQuality.Measured),
+                calculated: quantityQualities.Contains(QuantityQuality.Calculated)) switch
+            {
+                (missing: true, _, _, _) => CalculatedQuantityQuality.Missing,
+                (_, estimated: true, _, _) => CalculatedQuantityQuality.Estimated,
+                (_, _, measured: true, _) => CalculatedQuantityQuality.Measured,
+                (_, _, _, calculated: true) => CalculatedQuantityQuality.Calculated,
+                _ => throw new InvalidOperationException("Unknown quantity quality type"),
```
Comment (MWO1024) [inline source/Process.Application/Transactions/WholesaleServices/WholesaleServicesRequestAcceptedMapper.cs:184]: Is there a reason we do not use

                (missing: true, estimated: false, measured: false, calculated: false) => CalculatedQuantityQuality.Missing,
                (missing: true, _, _, _) => CalculatedQuantityQuality.Incomplete,
                (_, estimated: true, _, _) => CalculatedQuantityQuality.Estimated,
                (_, _, measured: true, _) => CalculatedQuantityQuality.Measured,
                (_, _, _, calculated: true) => CalculatedQuantityQuality.Calculated,
                _ => CalculatedQuantityQuality.NotAvailable,

as we do elsewhere?

File: source/IntegrationEvents/IntegrationEvents.Infrastructure/Factories/WholesaleServicesMessageFactory.cs [hunk 2]
```diff
@@ -47,7 +47,7 @@ public WholesaleServicesMessageFactory(IMasterDataClient masterDataClient)
             ActorRole.EnergySupplier,
             message.ChargeOwner,
             Guid.NewGuid(),
-            BusinessReasonMapper.Map(monthlyAmountPerChargeResultProducedV1.CalculationType),
+            BusinessReasonMapper.Map(monthlyAmountPerChargeResultProducedV1.CalculationType).Name,
```
Comment (RasmusGraabaek) [inline source/IntegrationEvents/IntegrationEvents.Infrastructure/Factories/WholesaleServicesMessageFactory.cs:50]: Do we save the code or name in the DB?

File: source/Process.Application/Transactions/WholesaleServices/WholesaleServicesRequestAcceptedMapper.cs [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                Parse(point.Price),
+                Parse(point.Amount)));
+
+            pointPosition++;
+        }
+
+        return points.AsReadOnly();
+    }
+
+    private static CalculatedQuantityQuality MapQuality(ReadOnlyCollection<QuantityQuality> quantityQualities)
```
Comment (MWO1024) [inline source/Process.Application/Transactions/WholesaleServices/WholesaleServicesRequestAcceptedMapper.cs:172]: Could we reuse `CalculatedQuantityQualityMapper`? And move it out of Aggregation?

File: source/Process.Domain/Transactions/WholesaleServices/ProcessEvents/EnqueuedAcceptedWholesaleServicesEvent.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Energinet.DataHub.EDI.OutgoingMessages.Interfaces.Models;
+
+namespace Energinet.DataHub.EDI.Process.Domain.Transactions.WholesaleServices.ProcessEvents;
+
+/// <summary>
+/// Event to raise when to enqueue accepted Wholesale Services response
+/// /// </summary>
```
Comment (RasmusGraabaek) [inline source/Process.Domain/Transactions/WholesaleServices/ProcessEvents/EnqueuedAcceptedWholesaleServicesEvent.cs:21]:
```suggestion
/// </summary>
```

File: source/Process.Application/Transactions/WholesaleServices/WholesaleServicesRequestAcceptedMapper.cs [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+            {
+                (missing: true, _, _, _) => CalculatedQuantityQuality.Missing,
+                (_, estimated: true, _, _) => CalculatedQuantityQuality.Estimated,
+                (_, _, measured: true, _) => CalculatedQuantityQuality.Measured,
+                (_, _, _, calculated: true) => CalculatedQuantityQuality.Calculated,
+                _ => throw new InvalidOperationException("Unknown quantity quality type"),
+            };
+    }
+
+    private static decimal? Parse(DecimalValue? input)
```
Comment (RasmusGraabaek) [inline source/Process.Application/Transactions/WholesaleServices/WholesaleServicesRequestAcceptedMapper.cs:188]: We have this here: namespace Energinet.DataHub.EDI.IntegrationEvents.Infrastructure.Protobuf;

File: source/Process.Application/Transactions/WholesaleServices/WholesaleServicesRequestAcceptedMapper.cs (hunk 3 above)
Comment (RasmusGraabaek) [inline source/Process.Application/Transactions/WholesaleServices/WholesaleServicesRequestAcceptedMapper.cs:172]: we have this here: namespace Energinet.DataHub.EDI.Process.Application.Transactions.Aggregations;

So we should move it "CalculatedQuantityQualityMapper"

**Review by RasmusGraabaek — APPROVED** [review]

**Pull request merged**